exposed-tests/src/main/kotlin/org/jetbrains/exposed/sql/tests/DatabaseTestsBase.kt

```diff
@@ -21,12 +21,7 @@ enum class TestDB(
     var db: Database? = null
 ) {
     H2({ "jdbc:h2:mem:regular;DB_CLOSE_DELAY=-1;" }, "org.h2.Driver"),
-    H2_MYSQL(
-        { "jdbc:h2:mem:mysql;MODE=MySQL;DB_CLOSE_DELAY=-1" }, "org.h2.Driver",
-        beforeConnection = {
-            Mode.getInstance("MySQL").convertInsertNullToZero = false
-        }
-    ),
+    H2_MYSQL({ "jdbc:h2:mem:mysql;MODE=MySQL;DB_CLOSE_DELAY=-1" }, "org.h2.Driver"),
```

Comment:
I believe this change is the reason for failing tests (as they all fails on H2_MYSQL)

Reply:
Note, that all tests are still run on v[1.4.199](https://github.com/JetBrains/Exposed/blob/3a9c08d04d4b8a4cfafc1149bc807347f31c65b5/buildSrc/src/main/kotlin/org/jetbrains/exposed/gradle/Versions.kt#L11), as no changes were made to the [Gradle test infrastructure setup](https://github.com/JetBrains/Exposed/blob/b3c6baef0e6d2025ec14c2d9515a927523195647/buildSrc/src/main/kotlin/org/jetbrains/exposed/gradle/DockerTestContainers.kt#L47)

Reply:
Should work with [1.4.200](https://github.com/JetBrains/Exposed/blob/b3c6baef0e6d2025ec14c2d9515a927523195647/buildSrc/src/main/kotlin/org/jetbrains/exposed/gradle/DockerTestContainers.kt#L47), by the way

Reply:
Hey @naftalmm, are you sure? I tried to bump H2 to 1.4.200, see https://github.com/JetBrains/Exposed/pull/1396

Reply:
As you can see in TC, these [particular failing tests](https://teamcity.jetbrains.com/viewLog.html?buildId=3712784&buildTypeId=Exposed_PullRequests) are passing with [1.4.200](https://teamcity.jetbrains.com/viewLog.html?buildId=3712702&buildTypeId=Exposed_PullRequests).

Reply:
They don't pass. Currently Exposed uses 1.4.199, not 1.4.200. 
1. In 1.4.200, as you can see in #1396, the `convertInsertNullToZero` option is removed: h2database/h2database#1942. Without this change, Exposed tests fail with `methodNotFound` at runtime.
2. After removing the dropped MySQL option, they [don't pass either](https://teamcity.jetbrains.com/buildConfiguration/Exposed_PullRequests/3712702?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true), because H2 will a NPE. [Fix is here](https://github.com/h2database/h2database/pull/3218), but won't be merged into H2 repo. 
=> 1.4.200 cannot be used.

Reply:
Look precisely at the [names of failing tests](https://teamcity.jetbrains.com/buildConfiguration/Exposed_PullRequests/3712702?buildTab=tests&status=failed) with 1.4.200. These [particular failing tests](https://teamcity.jetbrains.com/viewLog.html?buildId=3712784&buildTypeId=Exposed_PullRequests) are not among them. Although they are not among passed either. probably, because TC is configured in the way that they aren't even run if some more basic tests are failing, but you can run them locally and see they are passing (at least they are passing on my machine).

Yes, 1.4.200 cannot be used, just mentioned it spontaneously.

Reply:
Okay :D
But removing `convertInsertNullToZero` is still required, because its support is dropped in `202` too.

Reply:
@naftalmm Do the 4 failing test of this PR fail on your machine too? I can't reproduce it, on my machine (macOS with M1) they pass.

Reply:
Yes, these 4 tests are failing with default settings in `gradle.properties` (`dialect=none`) on my machine (Windows)

Reply:
Okay, because `dialect=none` uses 1.4.199. Could you try `dialect=h2_v2`?

Reply:
`dialect=h2_v2` ends up in dozens of failed tests (mostly because of missing runtime dependencies, but some are truly related to changes in the new version of H2). It happens because tests are still run for the [default set of DBs](https://github.com/JetBrains/Exposed/blob/e7148105ddc2fda9cc581ef2120c66f5f57d454a/exposed-tests/src/main/kotlin/org/jetbrains/exposed/sql/tests/DatabaseTestsBase.kt#L102) (not only for H2&H2_MySQL), but with `dialect=h2_v2` there are no [connectors for them](https://github.com/JetBrains/Exposed/blob/3a9c08d04d4b8a4cfafc1149bc807347f31c65b5/buildSrc/src/main/kotlin/org/jetbrains/exposed/gradle/DockerTestContainers.kt#L59-L61) in runtime.

Note, that `dialect=h2_v2` uses both 1.4.199 and 2.0.202 due to this [redundant codeline](https://github.com/JetBrains/Exposed/blob/3a9c08d04d4b8a4cfafc1149bc807347f31c65b5/buildSrc/src/main/kotlin/org/jetbrains/exposed/gradle/DockerTestContainers.kt#L49) (and God knows which classes are actually loaded in runtime).

I'd recommend starting researching the possibility of 2.0.202 support from the naive `Versions.h2` bump and seeing what obstacles are met and how to overcome them.
Support for both H2 versions is another task (mostly related to Gradle or/and TeamCity configuration).

Reply:
Gradle upgrades the dependency to the newer one, so with `dialect=2.0.202` the test always use H2 `2.0.202`. You could check it with `gradle exposed-test:dependencies`.
But indeed, I get 19 failed tests too.

Reply:
And yes, for integration TC config needs to be updated

Reply:
Now down to 14 failing test

Reply:
@naftalmm Yeah, you are right. For some reasons, without this line the exception handling is changed. You could try it at this clean branch: https://github.com/hfhbd/Exposed/commit/c68b76f3280f09e5be301f4036c255a22312a228
All currently failing tests with the old H2 version fails, because they expect some exception, which will not throw in the expected block, but later by H2. The current code block, which waits for the exception does not get it, and fails the test.

Reply:
After digging, the problem: `1.4.199` with MySQL does not throw an exception without this mode, is fixed in `1.4.200` and up. See sample project: https://github.com/hfhbd/testH2

Reply:
Yes, this code line is necessary for passing tests with 1.4.199. Just wrap it with `if (Constants.VERSION_MAJOR == 1 && Constants.VERSION_MINOR <= 4 && Constants.BUILD_ID <= 199)` check to make it work for all H2 versions

Reply:
Thanks! That's weird but works